File: Python/virtual-world-simulator/src/Entity/Animals/Wolf.py
import pygame
from src.Entity.Animals.Animal import Animal


class Wolf(Animal):

    def __init__(self, world, position):
        temp = pygame.image.load('assets/wolf.png')
        scaled = pygame.transform.scale(temp, (world.scale, world.scale))
        super().__init__(scaled, world, position, 'W', 9, 5)

    def collision(self, entity):
        if isinstance(entity, Wolf):
            breedPosition = self.breed()
            if breedPosition is not None:
                element = Wolf(self._world, breedPosition)
                self._world.setMapElement(breedPosition[0], breedPosition[1], element)
            return False

        return super().collision(entity)

File: C++/virtual-world-simulator/src/Entity/Animals/Wolf.cpp
#include "Wolf.h"

Wolf::Wolf(World* world, Point position) :
	Animal{ AssetManager::getAssetManager()->getAsset("wolf"), world, position, 'W', 9, 5 }
{
}

Wolf::Wolf(World* world, Point position, int lifeSpan, int strength) :
	Animal{ AssetManager::getAssetManager()->getAsset("wolf"), world, position, 'W', strength, 5, lifeSpan }
{
}

bool Wolf::collision(std::fstream& logFile, Entity& entity)
{
	Wolf* wolf = dynamic_cast<Wolf*>(&entity);
	if (wolf != nullptr)
	{
		this->breed<Wolf>(logFile, *wolf);
		return false;
	}

	return Animal::collision(logFile, entity);
}
